### ISSUE TITLE
Some Lifx bulbs are constantly being reported as offline then online, while other's are fine.

### ISSUE THREAD

**Issue by reefbarman** [devbobo/homebridge-lifx-lan#1]
Hi,

With the plugin enabled it finds all my bulbs but then some randomly cycle from online to offline and back again so siri can't always interact with them. Though the lifx app doesn't seem to have a problem! Have any clues?

**Comment by reefbarman** [comment]
Seems after some time they just get reported as offline and then I can't get them to come back online again using homebridge even though lifx app can still interact with them

**Comment by devbobo** [comment]
G'day @reefbarman,

Sorry for the delayed response, I've just gotten back from 2 weeks vacation.

This seems strange...I have lights that go offline from time to time, but when they are unavailable, they are also unavailable from the Lifx app.

The issue may be in the [node-lifx](https://github.com/MariusRumpf/node-lifx) library, you might want to ping @MariusRumpf.

Cheers,

David

**Comment by MariusRumpf** [comment]
Hi,

@devbobo node-lifx checks if a bulb is online by looking for an answer to the discovery requests send every 5 seconds. If any LIFX light did not answer to it for a period of discovery request it is set as offline. By default this value is set to three discoveries. This can be tweaked by using the `lightOfflineTolerance` setting for the `init` function.
The LIFX app has some special hacks and does detect offline/online lights different, but they are undocumented.

@reefbarman Do some light have a bad wifi connection maybe?

**Comment by reefbarman** [comment]
@MariusRumpf as far as I can determine the lights connection to the network
is fine (As mentioned I have no issue with the LIFX app ever) and the
lights themselves are no more then 4 meters away from the router in a small
flat

**Comment by devbobo** [comment]
@MariusRumpf thanks mate :+1: 

@reefbarman I've just [committed a change](https://github.com/devbobo/homebridge-lifx-lan/commit/3943d365f58d8fb8219c78429d93ca7cb0d152cd) that allows any of the node-lifx parameters to be passed from the config file. Can you give that a try and see if setting a higher value for `lightOfflineTolerance` helps ?

**Comment by devbobo** [comment]
closing this issue, as there has been no feedback for a few weeks.

**Comment by AndrewC-B** [comment]
I had this issue too. I progressively increased lightOfflineTolerance to 6 and found that I got substantially fewer offline/online messages appearing. I then increased lightOfflineTolerance to 10 and upped resendMaxTimes to 5 and now I get barely any offline/online disconnects (and when my light goes offline it seems to come online again quite quickly). Overall the experience is much more stable with these settings. So far, at least.
